spec/coordinates_spec.rb

```diff
@@ -0,0 +1,31 @@
+require_relative "../lib/coordinates"
+
+RSpec.describe Coordinates do
+  describe "#valid?" do
+    it "is true when the value is in the grid" do
+      described_class.valid_values.each do |value|
```

Comment:
I left a comment below about loops in tests and exhaustiveness, and looped back up to this spec. I don't think we need to test every valid coordinate. Also exposing the use of this class method/data feels bad. Typically I think of anything not tested as private, and this data feels like it doesn't belong in a test.

I think testing boundaries with a spec each makes more sense than testing every coordinate.

Reply:
One more thing about this spec that shows up in other places. I have found `subject` and also `described_class` to be much less clear than using the class constant itself. 

I like my tests to be completely unclever and duplicative so that it is clear what is happening.